### PR TITLE
Allow to pass custom columns as a prop

### PR DESCRIPTION
**Pull request by rvsia**
- also allow to disable the default ones

## columns

*array*

An array of columns definitions. They are merged with default columns by their keys.

## disableDefaultColumns

*boolean | string[]*

You can disable default columns. If you set this prop to `true`, then all the default columns are disabled. Or you can provide an array of keys to disable just specific ones: `[ 'display_name', 'last_seen' ]`.

**After**

```jsx
      disableDefaultColumns={['display_name']}
      columns={
          [{
              key: 'display_name_1',
              title: 'Different name',
              renderFunc: (name, id, { display_name }) => <h1>{display_name}</h1>
          }
          ]
      }
```

Screenshot 2021-04-30 at 11 41 33

### CODE REVIEW

**Comment by rvsia** [comment]
cc @bastilian

**Comment by bastilian** [comment]
Would it maybe make sense to just completely override the default columns when the columns prop is set?

Default columns could be made available to import and allow mixing the columns and also dictate their order.
We have had instances where UX/PM asked for a specific order or columns in tables including the systems/inventory table.

There is something along those lines implemented here: https://github.com/RedHatInsights/compliance-frontend/pull/1005/files#diff-e0494b62ef3cc7ba3308fa54b1933e5d3c299e030a4ac5adaef91ef23630a08dR10

**Comment by rvsia** [comment]
@bastilian So this is currently possible. Just set `disableDefaultFilters` and then import the default ones (they are available as export) and change them as you need in the columns.

File: packages/inventory/src/components/table/EntityTable.js [hunk 1]
```diff
@@ -62,6 +67,16 @@ const EntityTable = ({
...
 
+    const disabledColumns = Array.isArray(disableDefaultColumns) ? disableDefaultColumns : [];
+    const defaultColumnsFiltered = defaultColumns.filter(({ key }) =>
+        (key === 'tags' && showTags) || (key !== 'tags' && !disabledColumns.includes(key))
+    );
+
+    const columns = mergeArraysByKey([
+        typeof disableDefaultColumns === 'boolean' && disableDefaultColumns ? [] : defaultColumnsFiltered,
+        columnsProp || columnsRedux || []
+    ], 'key');
```
Comment (karelhala) [inline packages/inventory/src/components/table/EntityTable.js:78]: This is quite a lot of heavy computing, might be reasonable to move all of this `useEffect` where you could use the check from `useSelector` of `columnsRedux` and store the value in `ref`. We don't want to trigger yet another re-render, but the calculation could be usefull to moize this way. Or we can use `useMemo`. But the memoize function would be a bit complex.

Either way it should be reasonable to somehow limit the amount of calculations for this piece of code as it uses `filter` and `mergeArraysByKey` at each re-render.

**Review by karelhala — COMMENTED** [review]
Looking hella good! Just a bit of optimization would be nice.

File: packages/inventory/src/components/table/EntityTable.js (hunk 1 above)
Comment (rvsia) [reply packages/inventory/src/components/table/EntityTable.js]: done!

**Review by karelhala — APPROVED** [review]

**Comment by AllenBW** [comment]
Any update on when we might see this work in prod beta/stable?

**Comment by karelhala** [comment]
@AllenBW we are releasing chrome to all envs this sprint, so it should land in prod-stable by end of this week in best case scenario.

**Comment by AllenBW** [comment]
ah! great news, thanks for the update 🤗

**Pull request merged**